src/pages/Messages.jsx

```diff
@@ -1,23 +1,97 @@
-import { Box, Typography } from "@mui/material";
-import React from "react";
+import {
+  Box,
+  IconButton,
+  InputAdornment,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import HighlightOffIcon from "@mui/icons-material/HighlightOff";
+import React, { useState } from "react";
+import MessageBoard from "../components/MessageBoard"; 
+
+const chatsRes = [
+  {
+    name: "Festus",
+    imgSrc: "./images/festus.jpg",
+    lastMessage: "Hey, How are you?",
+    timestamp: "12:01",
+    senderId: 1,
+  },
+  {
+    name: "Charlie",
+    imgSrc: "./images/example-1.jpeg",
+    lastMessage: "Hey, How are you?",
+    timestamp: "11:10",
+    senderId: 2,
+  },
+];
 
 const Messages = () => {
+  const [chats, setChats] = useState(chatsRes);
+  const [searchText, setSearchText] = useState("");
+
+  const handleSearchTextChange = (event) => {
+    if (!event) {
+      setChats(chatsRes);
+      return;
+    }
+    const newText = event.target.value;
+    setSearchText(newText);
+    const filteredChats = chatsRes.filter((chat) =>
+      chat.name.toLowerCase().includes(newText.toLowerCase())
+    );
```

Comment:
It would be great if this could search through messages

Reply:
Is this easy to support within this PR? Or should we leave it out?

Reply:
@charliefischer

Reply:
ummmm maybe do it in the next PR - chatting to Ricky I think this could be good if I tidy up any bigger issues, release it and then he can copy the few lines of firestore retrieval to add to the profile pages\